src/types/basic.ts

```diff
@@ -1,7 +1,182 @@
-export interface EnvVarObject {
-	[name: string]: string;
-}
+import * as t from 'io-ts';
+import { chain } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/function';
 
-export interface LabelObject {
-	[name: string]: string;
-}
+/**
+ * A short string is a non null string between
+ * 0 and 255 characters
+ */
+export const ShortString = new t.Type<string, string>(
+	'ShortString',
+	(i: unknown): i is string => t.string.is(i) && i.length <= 255,
+	(i, c) =>
+		pipe(
+			t.string.validate(i, c),
+			chain((s) =>
+				s.length <= 255
+					? t.success(s)
+					: t.failure(s, c, 'must be at most 255 chars long'),
+			),
+		),
+	t.identity,
+);
+
+// Note: assigning this type to a string will not throw compilation errorrs.
+//
+// e.g. the following will compile without issues.
+// ```
+// const x: ShortString = 'a'.repeat(300);
+// ```
+export type ShortString = t.TypeOf<typeof ShortString>;
+
+/**
+ * A string identifier is a string that encodes a
+ * positive integer (an id to be used as a database id)
+ *
+ * e.g.
+ * Invalid decimal strings: 'aaa', '0xaaa'
+ * Valid decimal strings: '0', '123'
+ */
+export const StringIdentifier = new t.Type<string, string>(
+	'StringIdentifier',
+	(i: unknown): i is string =>
+		t.string.is(i) && !isNaN(+i) && +i === parseInt(i, 10) && +i >= 0,
+	(i, c) =>
+		pipe(
+			t.string.validate(i, c),
+			chain((s) =>
+				!isNaN(+s) && +s === parseInt(s, 10) && +s >= 0
```

Comment:
Why is `!isNaN(+1) && +i === parseInt(i, 10) && +i >= 0` (or `s` instead of `i` validated twice?

Reply:
That's just the way [io-ts custom types work](https://github.com/gcanti/io-ts/blob/master/index.md#custom-types) one is the typescript guard and the second one is the actual validator